crates/bevy_ecs/src/schedule/state.rs

```diff
@@ -137,6 +138,16 @@ impl<S: States> NextState<S> {
     }
 }
 
+/// Event sent when any state transition of `S` happens.
+/// If you know exactly what state you want to respond to ahead of time, consider [`OnEnter`], [`OnTransition`], or [`OnExit`]
```

Comment:
```suggestion
/// Event sent when any state transition of `S` happens.
///
/// If you know exactly what state you want to respond to ahead of time, consider [`OnEnter`], [`OnTransition`], or [`OnExit`]
```

Just to separate the paragraphs.